src/pineko/scale_variations.py

```diff
@@ -0,0 +1,237 @@
+"""Module to generate scale variations."""
+import pathlib
+
+import numpy as np
+import pineappl
+import rich
+from eko import beta
+
+from . import check
+
+
+def sort_orders(order):
+    """Define the sorting for an order. In particular the order is sorted according to the as order."""
+    return order[0]
+
+
+def ren_sv_coeffs(m, max_as, logpart, which_part, nf):
+    """Return the ren_sv contribution relative to the requested log power and perturbative order contribution (which_part).
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    max_as : int
+        max order of alpha_s
+    logpart : int
+        power of the renormalization scale log asked
+    which_part : int
+        asked perturbative order contribution to be rescaled
+    nf : int
+        number of active flavors
+
+    Returns
+    -------
+    float
+        renormalization scale variation contribution
+    """
+    pifactor = 1.0 / (4.0 * np.pi)
+    if max_as == 0:
+        return 0.0
+    elif max_as == 1:
+        return -(-m * beta.beta_qcd((2, 0), nf) * pifactor)
+    elif max_as == 2:
+        if which_part == 0:
+            if logpart == 1:
+                return -m * beta.beta_qcd((3, 0), nf) * (pifactor**2)
+            else:
+                return (
+                    0.5
+                    * m
+                    * (m + 1)
+                    * (beta.beta_qcd((2, 0), nf) ** 2)
+                    * (pifactor**2)
+                )
+        else:
+            return -(m + 1) * beta.beta_qcd((2, 0), nf) * pifactor
+
+
+def compute_scale_factor(m, nec_order, to_construct_order, nf):
+    """Compute the factor of renormalization scale variation.
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order
+    nec_order : tuple(int)
+        tuple of the order that has to be rescaled to get the scale varied order
+    to_contruct_order : tuple(int)
+        tuple of the scale varied order to be constructed
+    nf : int
+        number of active flavors
+
+    Returns
+    -------
+    float
+        full contribution of ren sv
+    """
+    max_as = to_construct_order[0] - m
+    logpart = to_construct_order[2]
+    return ren_sv_coeffs(m, max_as, logpart, nec_order[0] - m, nf)
+
+
+def compute_orders_map(m, max_as, min_al):
+    """Compute a dictionary with all the necessary orders to compute to have the full renormalization scale variation.
+
+    Parameters
+    ----------
+    m : int
+        first non zero perturbative order of the grid
+    max_as : int
+        max alpha_s order
+    min_al : int
+        al order of leading order
+    Returns
+    -------
+    dict(tuple(int))
+        description of all the needed orders
+    """
+    orders = {}
+    for delt in range(max_as):
+        orders[(m + max_as, min_al, delt + 1, 0)] = [
+            (m + de, min_al, 0, 0) for de in range(max_as - delt)
+        ]
```

Comment:
Re the second point: not a problem from UI point of view, calling once or twice might even be fine.

Simply, I do not see much the point of computing one and not the other, I believe we should compute all the missing ones. Is there a use case for computing only NNLO, if NLO are also missing?

Reply:
> Simply, I do not see much the point of computing one and not the other, I believe we should compute all the missing ones. Is there a use case for computing only NNLO, if NLO are also missing?

No of course but if you want to include all the sv terms up the one the user is asking, it means that we need to be careful and testing that all the centrals are included (I mean the ones used to compute the sv terms). Moreover, if I don't see when, I believe it could be useful to have the possibility of including only one contribution even in relation with the k_factor business that, in principle can also be used to include sv k_factors.